flexmeasures/cli/db_ops.py

```diff
@@ -122,18 +135,47 @@ def restore(file: str):
         % flexmeasures db-ops restore FILE
 
     """
-
     db_uri: str = app.config.get("SQLALCHEMY_DATABASE_URI")  # type: ignore
     db_host_and_db_name = db_uri.split("@")[-1]
     click.echo(f"Restoring {db_host_and_db_name} database from file {file}")
-    command_for_restoring = f"pg_restore -d {db_uri} {file}"
+    if pg_version is None:
+        pg_version = try_getting_postgres_version(db_uri)
+    command_for_restoring = (
+        f"pg_restore {pg_cluster_arg_from_version(pg_version)} -d {db_uri} {file}"
+    )
     try:
-        subprocess.check_output(command_for_restoring, shell=True)
+        subprocess.check_output(
+            command_for_restoring, shell=True, stderr=subprocess.STDOUT, text=True
+        )
+    except subprocess.CalledProcessError as e:
+        click.secho(f"Exception happened during restore: {e.output}", **MsgStyle.ERROR)
+        click.secho("db restore unsuccessful", **MsgStyle.ERROR)
+    else:
         click.secho("db restore successful", **MsgStyle.SUCCESS)
 
-    except Exception as e:
-        click.secho(f"Exception happened during restore: {e}", **MsgStyle.ERROR)
-        click.secho("db restore unsuccessful", **MsgStyle.ERROR)
+
+def pg_cluster_arg_from_version(pg_version: str | None) -> str:
+    """
+    Small utility, turning the target postgres version into an arg for pg utils,
+    so that they know what they're supposed to work against.
+    More info (e.g. on the format) at https://manpages.ubuntu.com/manpages/trusty/man1/pg_wrapper.1.html
+    """
+    if pg_version:
+        return f"--version {pg_version}"
```

Comment:
Indeed, I failed to properly validate my solution. The error I get with `return f"--cluster {pg_version}/main"` is:
```
Exception happened during dump: Error: Invalid version 15.5 specified with --cluster
```

I'll revert my last commit. Then, we can try to work the issue I encounter with v13 (any suggestions?), or we could document that it only works for v14 or higher.

Reply:
Actually, according to the [PostgreSQL docs (v13)](https://www.postgresql.org/docs/13/app-pgdump.html):

Because pg_dump is used to transfer data to newer versions of PostgreSQL, the output of pg_dump can be expected to load into PostgreSQL server versions newer than pg_dump's version. pg_dump can also dump from PostgreSQL servers older than its own version. (Currently, servers back to version 8.0 are supported.) However, **pg_dump cannot dump from PostgreSQL servers newer than its own major version; it will refuse to even try, rather than risk making an invalid dump.** Also, it is not guaranteed that pg_dump's output can be loaded into a server of an older major version — not even if the dump was taken from a server of that version.

Docs of newer versions have the same text, so I wonder how this could be working for you?

Reply:
Might be interesting. Maybe using Docker pg containers for development makes more sense in general.

Do you propose this both for `restore` and `dump`?

I want to restore into my local database.

Reply:
> Do you propose this both for `restore` and `dump`?

Yes, but only in the context of dumping from and restoring to a (newer) remote database server. For restoring to an older local database server, I'm not sure whether going with the docker-route would help with anything.